freshmaker/lightblue.py

```diff
@@ -1579,5 +1579,249 @@ def _get_images_to_rebuild(image):
+                    },
+                    {
+                        "$or": [
+                            {
+                                "field": "content_sets.*",
+                                "op": "=",
+                                "rvalue": content_set
+                            }
+                            for content_set in content_sets
+                        ]
+                    },
+                    {
+                        "field": "repositories.*.published",
+                        "op": "=",
+                        "rvalue": True,
+                    },
+                ]
+            },
+            # Start with a small projection and increase it once a fixed image is found by
+            # querying by the NVR with the default projection
+            "projection": [
+                {"field": "brew.build", "include": True},
+                {
+                    "field": "rpm_manifest.*.rpms",
+                    "include": True,
+                    "match": {
+                        "$or": [
+                            {
+                                "field": "name",
+                                "op": "=",
+                                "rvalue": rpm_name
+                            } for rpm_name in rpm_name_to_nvrs.keys()
+                        ]
+                    },
+                    "project": [
+                        {"field": "nvra", "include": True},
+                        {"field": "name", "include": True},
+                    ]
+                },
+                {"field": "repositories.*.repository", "include": True, "recursive": True},
+                {"field": "content_sets", "include": True, "recursive": True},
+            ]
+        }
+        images = self.find_container_images(request)
+        if not images:
+            log.error("Could not find an image with the name and version of %s-%s", name, version)
+            return
+
+        candidate_images = []
+        for image in images:
+            # If it's not on the same repositories or the regex matched something unexpected, then
+            # skip it
+            candidate_image_group = self.describe_image_group(image)
+            if candidate_image_group != image_group:
+                log.debug(
+                    "The image %s did not have the correct image group (%s != %s)",
+                    image.nvr,
+                    candidate_image_group,
+                    image,
+                )
+                continue
+
+            # Due to filtering by installed RPMs taking too long in lightblue, perform the filter
+            # here since the projection (returned RPM manifest from lightblue) has the filtering
+            # applied. This is to be conservative in the event a child image relies on the RPM but
+            # it is no longer installed
+            if {rpm["name"] for rpm in image.get_rpms() or []} != rpm_name_to_nvrs.keys():
```

Comment:
An image doesn't need to have all the rpms from the cve fix rpm advisory installed. We can remove this check.

Reply:
@qixiang `rpm_name_to_nvrs` just contains the RPMs that were present in the vulnerable parent that were fixed by the advisory. So this check verifies that those same RPMs are still present in the fixed version of the parent image. This is to ensure that in the event the child image relies on those RPMs already being installed by the parent image, that they continue to be installed.

I admit that it's very cautious, but this is what @gnaponie and I had agreed to when I discussed this with her a few weeks ago.

@gnaponie do you still agree? I can remove this check if you'd like.

Reply:
I see, I misread it, I thought the `rpm_name_to_nvrs` are rpms from the rpm advisory. I agree it's ok to keep this check.